### ISSUE TITLE
Wrapping an undeclared variable breaks valid code above

### ISSUE THREAD

**Issue by ElianCordoba** [microsoft/TypeScript#48000]
# Bug Report

### 🔎 Search Terms

- Missing semicolon
- Error recovery
- undeclared

### 🕗 Version & Regression Information

Tested all the way back to version 2.0 and it still occurs. Still happens in nightly

- This is the behaviour in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

[Code](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220215&ssl=8&ssc=6&pln=1&pc=1#code/PTAEBUHtQWwQwNYFNQBcAWKkCduW6AObRwDucAngFwBQIoAtKAIIAmrocoAzkjAJYBjSABtIAOzTQR-cSgBsdMEwBKfSADcUABzjYk4jEm79uPAK64JrWYVAAjPUsagAIkgBmslI+w0awuLcqKAekNAAvKAA2gC6-gAUvgCUQA)

### 💻 Code

```ts
// To make the error go away:
// - Add a semicolon to line 6
// - Remove parenthesis surronding bar
// - Define bar

const foo = []

(bar)
```

### 🙁 Actual behavior

The error message it's confusing. The compiler gives you the hint of adding a semicolon though

### 🙂 Expected behavior

No error should be shown beside the undeclared variable one

**Comment by ElianCordoba** [comment]
Lucky number 48000 :)

**Comment by fatcerberus** [comment]
This is an ASI (automatic semicolon insertion) hazard - the code is parsed as `const foo = [](bar)`.  It doesn't matter whether `bar` is define beforehand or not - this code, as written, isn't going to do what you think it does.

**Comment by fatcerberus** [comment]
> Is it feasible to do some experience enhance magic to prevent developing shooting their own feet with this?

That's why you get this error:

```
This expression is not callable.
  Type 'never[]' has no call signatures.(2349)
input.tsx(6, 13): Are you missing a semicolon?
```

The code is trying to call `[](bar)` as if `[]` were a function, and since that makes no sense, TS suggests adding a semicolon.
